File: src/main/kotlin/space/davids_digital/kiri/rest/dto/filesystem/FileSystemNodeDto.kt
package space.davids_digital.kiri.rest.dto.filesystem

import java.time.OffsetDateTime
import java.util.*

data class FileSystemNodeDto(
    val id: UUID,
    val spaceId: UUID,
    val parentId: UUID?,
    val name: String,
    val type: Type,
    val mimeType: String?,
    val size: Long,
    val attributes: Map<String, String>,
    val createdAt: OffsetDateTime,
    val updatedAt: OffsetDateTime,
) {
    enum class Type {
        DIRECTORY,
        FILE,
    }
}

File: admin-ui/lib/api/types/filesystem/FileSystemNodeDto.ts
export type FileSystemNodeDto = {
    id: string;
    spaceId: string;
    parentId: string | null;
    name: string;
    type: FileSystemNodeDto.Type;
    mimeType: string;
    size: number;
    attributes: { [ ket: string ]: string };
    createdAt: number;
    updatedAt: number;
}

export namespace FileSystemNodeDto {
    export type Type = "directory" | "file";
}